python/sglang/srt/layers/communicator.py

```diff
@@ -422,6 +420,12 @@ def _gather_hidden_states_and_residual(
                 dp_scatter(residual, hidden_states, forward_batch)
                 if hidden_states.shape[0] != 0:
                     hidden_states = layernorm(hidden_states)
+            else:
+                # CommunicateSummableTensorPairFn._scatter_hidden_states will
+                # copy from global_hidden_states to hidden_states. If we don't
+                # have this clone, they will both point to gathered_buffer.
+                # Normally, layernorm would perform this clone.
+                hidden_states = hidden_states.clone()
```

Comment:
Can we use `torch.empty_like(forward_batch.gathered_buffer)` to avoid this clone? You may refer to #7705

Reply:
I found that https://github.com/sgl-project/sglang/pull/8539 fixes this issue by avoiding dp_scatter, so this fix won't be necessary.

Reply:
Pushed a fix with `empty_like()`. I prefer to do this in `_scatter_hidden_states` because the new buffer we create is smaller than if we were to do in line 414. We also only need to create the new buffer if the storage is the same which is only true for a few cases